salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+
+        Arguments:
+            client {hvac} -- hvac client
+            ret {dict} -- salt state result
+
+        Returns:
+            [list] -- policies
+        """
+        log.info('Retrieving policies from vault...')
+        polices = []
+        try:
+            policies_resp = client.sys.list_policies()
+
+            for policy in policies_resp['data']['policies']:
+                if not (policy == 'root' or policy == 'default'):
+                    polices.append(policy)
+
+        except Exception:
+            raise
+
+        log.info('Finished retrieving policies from vault.')
+
+        return polices
+
+    def push_policies(self, client, remote_policies, local_policies, ret):
+        """Push policies from local config to remote vault server
+
+        Arguments:
+            client {hvac} -- hvac client
+            remote_policies {list} -- policies from the remote vault server
+            local_policies {list} -- policies from local config
+            ret {dict} -- salt state result
+        """
+        log.info('Pushing policies from local config folder to vault...')
+        new_policies = []
+        try:
+            for policy in local_policies:
+                client.sys.create_or_update_policy(
+                    name=policy['name'],
+                    policy=policy['content']
+                )
+                if policy['name'] in remote_policies:
+                    log.debug('Policy "%s" has been updated.', policy["name"])
+                else:
+                    new_policies.append(policy["name"])
+                    log.debug('Policy "%s" has been created.', policy["name"])
+
+            # Build return object
+            ret['changes']['old'] = remote_policies
+            ret['changes']['new'] = new_policies or "No changes"
+        except Exception:
+            raise
+
+        log.info('Finished pushing policies local config folder to vault.')
+
+    def cleanup_policies(self, client, remote_policies, local_policies, ret):
+        """Removes policies that are not present in the local config
+
+        Arguments:
+            client {hvac} -- hvac client
+            remote_policies {list} -- policies current on the remote vault server
+            local_policies {list} --policies from local config
+            ret {dict} -- salt state result
+        """
+        log.info('Cleaning up vault policies...')
+        has_change = False
+        try:
+            for policy in remote_policies:
+                if policy not in [pol['name'] for pol in local_policies]:
+                    log.debug(
+                        '"%s" is not found in configs folder. Removing it from vault...', policy)
+                    has_change = True
+                    client.sys.delete_policy(name=policy)
+                    log.debug('"%s" is removed.', policy)
+
+            if has_change:
+                ret['changes']['new'] = [ob['name'] for ob in local_policies]
+        except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.